replica/table.cc

```diff
@@ -1027,12 +1027,24 @@ void table::do_trigger_compaction() {
 }
 
 void table::trigger_offstrategy_compaction() {
+    (void)perform_offstrategy_compaction().then_wrapped([this] (future<bool> f) {
+        if (f.failed()) {
+            auto ex = f.get_exception();
+            tlogger.warn("Offstrategy compaction of {}.{} failed: {}, ignoring", schema()->ks_name(), schema()->cf_name(), ex);
+        }
+    });
+}
+
+future<bool> table::perform_offstrategy_compaction() {
     // If the user calls trigger_offstrategy_compaction() to trigger
     // off-strategy explicitly, cancel the timeout based automatic trigger.
     _off_strategy_trigger.cancel();
-    if (!_maintenance_sstables->all()->empty()) {
-        _compaction_manager.submit_offstrategy(this);
+    if (_maintenance_sstables->all()->empty()) {
+        return make_ready_future<bool>(false);
     }
+    return _compaction_manager.perform_offstrategy(this).then([] {
+        return make_ready_future<bool>(true);
+    });
```

Comment:
looks like can be simplified with coroutine.

Reply:
Fixed in v2